examples/token-classification/infer.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import time
+import torch
+import logging
+from PIL import Image, ImageDraw, ImageFont
+from datasets import load_dataset
+from transformers import LiltForTokenClassification, LayoutLMv3FeatureExtractor, AutoTokenizer
+from optimum.habana import GaudiConfig
+import habana_frameworks.torch.core as htcore
+
+logger = logging.getLogger(__name__)
+
+def set_device(device_type):
+    if device_type == "cuda":
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    elif device_type == "hpu":
+        device = torch.device("hpu")
+    else:
+        device = torch.device("cpu")
+    return device
+
+def unnormalize_box(bbox, width, height):
```

Comment:
You mean to add `dtype` for the `unnormalize_box` function? I don't agree that this function needs `dtype`.

The `unnormalize_box` function doesn't require `dtype` because it operates directly on numerical inputs (integers or floats) without needing to explicitly specify or convert data types. Here’s a closer look at what the function does and why `dtype` isn’t necessary:

```python
def unnormalize_box(bbox, width, height):
    return [
        width * (bbox[0] / 1000),
        height * (bbox[1] / 1000),
        width * (bbox[2] / 1000),
        height * (bbox[3] / 1000),
    ]
```

- **bbox**: A list or tuple of four numbers (assumed to be integers or floats) representing normalized bounding box coordinates.

The function scales the normalized bounding box coordinates (`bbox`) to the actual image size using simple arithmetic operations. Each coordinate in `bbox` is divided by 1000 and then multiplied by the respective dimension (`width` or `height`). There is no need to convert data types within the function because the input types (integers or floats) are directly compatible with the arithmetic operations performed. Python inherently handles numerical types flexibly. When performing arithmetic operations like division and multiplication, the resulting type is automatically managed (e.g., division of integers results in a float).

Even with PyTorch's tensors, this would be handled on the dtype device themselves without any required change of device type. It doesn't make sense to change `dtype` on these simple math operations either. So this is handled implicitly.